### PR TITLE
docs(proccess): add versioning and maintenance policy

### LINKED ISSUE #3681: Maintenance: Create maintenance and versioning policy guide
### Why is this needed?

We need to create a comprehensive guide to inform our customers about our policies regarding the maintenance and versioning of Powertools for AWS Lambda.

### Which area does this relate to?

Other

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [X] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #3681 

### Changes

In this pull request, we are adding a comprehensive guide to inform our customers about our policies regarding the maintenance and versioning of Powertools or AWS Lambda. 

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [ ] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: docs/versioning.md
Comment (leandrodamascena) [inline docs/versioning.md:6]: Please reenable `MD013` before merging it.

**Comment by heitorlessa** [comment]
Thank you @leandrodamascena! Starting the editing process.

File: docs/versioning.md
Comment (heitorlessa) [reply docs/versioning.md:6]: suggestion: keep it to prevent changing pre-approved legal wording.

**Comment by heitorlessa** [comment]
Pushed the first editing pass. Import notes to discuss on a call as details matter.

---

**Significant contract abiding worth reviewing**

* No mention of the term `semantic versioning`
* GA versions minimum support of 2 years (24 months)
  * _suggestion: bind to Lambda runtime instead_
* At least 6 months announcement for maintenance releases
* Maintenance versions minimum support of 1 year
  * _suggestion: given staffing and EOL supply chain, we may not be able to do it._
* End of support versions are kept on GitHub _(e.g., v2 branch)_
  * _suggestion: keep package managers only to prevent higher CI charges on checkout (sparse checkout is possible)_ 
* GA versions support underlying EOL dependencies for at least 6 months
  * _suggestion: same issue as maintenance versions for 1yr_

## Questions

**Q: Do we want to keep EOL versions source code on GitHub?**

**Q: Do we want to support GA releases for a minimum of 2 years?**

**Q: Are we comfortable supporting underlying EOL dependencies for a minimum of 6 months (e.g., Pydantic v2, Middy 4, etc.)?**

**Q: Do we want to bundle EOL documentation versions _(1.x)_, keep them _(2.32.0)_?**

**Comment by heitorlessa** [comment]
Posting update after a call between @leandrodamascena @rubenfonseca and myself.

## Answered questions

**Q: Do we want to keep EOL versions source code on GitHub?**

We will keep as part of the Git tree e.g., tags, commits, etc. We will NOT keep a separate branch for each EOL version _e.g., v1_.

**Q: Do we want to support GA releases for a minimum of 2 years?**

Yes, as we've already done so historically.

**Q: Are we comfortable supporting underlying EOL dependencies for a minimum of 6 months (e.g., Pydantic v2, Middy 4, etc.)?**

No. We wouldn't have the know-how to vendor and backport each 3P critical fix dependency -- it can easily become a supply chain security issue given the plurality of each 3P community.

**Q: Do we want to bundle EOL documentation versions _(1.x)_, keep them _(2.32.0)_?**

Keep as-is. Documentation rebuilds are taking approximately 1m10s end-to-end with the new architecture _(e.g., S3 over GitHub Pages)_.

File: docs/versioning.md
Comment (rubenfonseca) [inline docs/versioning.md]: Powertools for AWS Lambda follows **the**

Comment (rubenfonseca) [inline docs/versioning.md]: This means we will stop supporting deprecated Language Runtime**s** without increasing the major SDK version.

Comment (heitorlessa) [reply docs/versioning.md]: yes. I'll make a minor grammar correction here.

**Comment by heitorlessa** [comment]
gonna try to meet @am29d @dreamorosi and @hjgraca to hear their thoughts from TS.

File: docs/versioning.md
Comment (dreamorosi) [inline docs/versioning.md]: minor suggestion: change to Middy.js to use official naming

**Review by dreamorosi — APPROVED** [review]
Other than the minor comment I've left on the name of one of the 3rd party dependencies, I agree with the document and its content as discussed during the sync.

Great work to all who contributed to its redaction, this will be very useful for both maintainers and customers.

File: docs/versioning.md
Comment (leandrodamascena) [reply docs/versioning.md]: Done

Comment (heitorlessa) [reply docs/versioning.md]: thank you :) fixing shortly.

**Comment by heitorlessa** [comment]
Good to go now @leandrodamascena -- feel free to merge and refresh the latest documentation!

**Review by rubenfonseca — APPROVED** [review]

**Pull request merged**
